analyses/survival-analysis/02-summary-statics-TP53.Rmd

```diff
@@ -0,0 +1,268 @@
+combined_filtered <- combined %>%
+  filter(binned_age %in% group_summary$binned_age[group_summary$count >= 3])
+
+ggplot(combined_filtered, aes(x = factor(binned_age, levels = c("[0,15]", "(15,26]", "(26,40]", "(40,62]", ">62", "not reported")), y = tp53_score)) + 
+  geom_boxplot() +
+  geom_jitter(aes(color = broad_mol), width = 0.3, alpha = 0.5) + 
+  stat_compare_means(comparisons = list(c("[0,15]", "(15,26]"), c("[0,15]", "(26,40]"), c("[0,15]", "(40,62]"), c("[0,15]", ">62"))) + 
+  xlab("") + 
+  theme_Publication()
+
+```
+## save the plot
+
+```{r}
+ggsave(file.path(plots_dir, "boxplot_tp53_score_binned_age.png"))
+
+```
+
+```{r}
+p1 <- ggplot(combined, aes(x = reported_gender, y = tp53_score)) + 
+      geom_boxplot() +
+      geom_jitter(aes(color = broad_mol), width = 0.3, alpha = 0.5) + 
+      stat_compare_means(comparisons = list(c("Female", "Male"))) + 
+      ggtitle(label = "All samples") + 
+      xlab("") + 
+      theme_Publication()
+
+p2 <- ggplot(HGG_survival, aes(x = reported_gender, y = tp53_score)) + 
+      geom_boxplot() +
+      geom_jitter(width = 0.3, alpha = 0.5) + 
+      stat_compare_means(comparisons = list(c("Female", "Male"))) + 
+      ggtitle(label = "HOPE cohort") + 
+      xlab("") + 
+      theme_Publication()
+
+p3 <- ggplot(GBM_df, aes(x = reported_gender, y = tp53_score)) + 
+      geom_boxplot() +
+      geom_jitter(width = 0.3, alpha = 0.5) + 
+      stat_compare_means(comparisons = list(c("Female", "Male"))) + 
+      ggtitle(label = "GBM") + 
+      xlab("") + 
+      theme_Publication()
+
+p1 + p2 + p3
+
+```
+
+## save the plot
+
+```{r}
+ggsave(file.path(plots_dir, "boxplot_tp53_score_gender.png"), height = 6, width = 12)
+
+```
+
+
+## The distribution of TP53 score among different races 
+
+```{r}
+group_summary <- HGG_survival %>%
+  group_by(race) %>%
+  summarize(count = n())
+
+# Filter the groups with at least 3 samples
+HGG_filtered <- HGG_survival %>%
+  filter(race %in% group_summary$race[group_summary$count >= 3])
+
+p1 <- HGG_filtered %>% 
+  mutate(race = fct_reorder(race, -tp53_score, .fun='median' )) %>%
+  ggplot(aes(x = race, y = tp53_score)) + 
+  geom_boxplot() +
+  geom_jitter(width = 0.3, alpha = 0.5) + 
+  xlab("") +
+  stat_compare_means(comparisons = list(c("Asian", "Reported Unknown"), c("Asian", "Black or African American"), c("Asian", "White"), c("Asian", "Other"))) +
+  theme_Publication() + 
+  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
+
+p1
+```
+
+### save the plot
+
+```{r}
+ggsave(file.path(plots_dir, "boxplot_tp53_score_race.png"))
+
+```
+
+### tp53 score distribution about different molecular subtypes 
+
+```{r}
+combined_filtered <- combined %>% 
+  filter(!is.na(tp53_score)) 
+
+p1 <- combined_filtered %>% 
+  mutate(broad_mol = fct_reorder(broad_mol, -tp53_score, .fun='median')) %>%
+  ggplot(aes(x = broad_mol, y = tp53_score)) + 
+  geom_jitter(aes(color = tp53_altered), width = 0.3, alpha = 0.5) +
+  geom_boxplot() +
```

Comment:
Could you swap these two lines as in the other plots, so the boxplots don't cover the points?

Reply:
Good catch! Change has been made